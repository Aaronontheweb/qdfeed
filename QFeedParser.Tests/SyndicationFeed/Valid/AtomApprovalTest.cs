using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;
using QDFeedParser.Tests.Extensions;

namespace QDFeedParser.Tests.SyndicationFeed.Valid
{
    [TestFixture]
    public class AtomApprovalTest
    {
        protected IFeedFactory Factory;
        protected Uri feeduri;
        protected int itemCount;

        [SetUp]
        public void SetUp()
        {
            this.Factory = new FileSystemFeedFactory();
            feeduri = new Uri(TestFileLoader.LoadValidAtomTestCases(TestFileLoader.TestFileType.FileSys).Last().Arguments[0].ToString());
            itemCount = 3;
        }

        [UseReporter(typeof(DiffReporter))]
        [Test, Description("Uses approvals to determine if the output from the feed entry parsing is correct or not.")]
        public virtual void ApproveAtomFeedItemParse()
        {
            IFeed feed = Factory.CreateFeed(feeduri);
            var strFeedItems = from feeditem in feed.Items
                               select feeditem.ToApprovalString();
            Approvals.Approve(strFeedItems, "feed items");
        }

        [UseReporter(typeof(DiffReporter))]
        [Test, Description("Uses approvals to determine if the GetItems function produces the correct amount of feed entries.")]
        public virtual void ApproveAtomFeedItemExplicitCountParse()
        {
            var feed = Factory.CreateFeed(feeduri);
            var strFeedItems = from feeditem in feed.Items.Take(itemCount)
                               select feeditem.ToApprovalString();
            Approvals.Approve(strFeedItems, "feed items");
        }

    }
}
