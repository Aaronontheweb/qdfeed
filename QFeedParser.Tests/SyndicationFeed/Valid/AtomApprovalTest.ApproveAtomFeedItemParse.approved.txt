feed items[0] = Title: SmartDraw 2010 Guided Tour
Published: 5/31/2010 2:48:53 PM
URL: http://www.youtube.com/watch?v=NJPrllhYZrg&feature=youtube_gdata
Author: smartdraw
Id: tag:youtube.com,2008:video:NJPrllhYZrg
Content: 
Categories: 

feed items[1] = Title: How to Manage a Project with SmartDraw
Published: 5/31/2010 1:59:14 PM
URL: http://www.youtube.com/watch?v=54AHBOA1jJ0&feature=youtube_gdata
Author: smartdraw
Id: tag:youtube.com,2008:video:54AHBOA1jJ0
Content: 
Categories: 

feed items[2] = Title: How to Document and Improve Processes with SmartDraw
Published: 5/30/2010 5:25:38 AM
URL: http://www.youtube.com/watch?v=SfYjkcQofaA&feature=youtube_gdata
Author: smartdraw
Id: tag:youtube.com,2008:video:SfYjkcQofaA
Content: 
Categories: 

feed items[3] = Title: How to Make Presentations with SmartDraw
Published: 5/31/2010 10:29:13 AM
URL: http://www.youtube.com/watch?v=tn5k0sFUgnw&feature=youtube_gdata
Author: smartdraw
Id: tag:youtube.com,2008:video:tn5k0sFUgnw
Content: 
Categories: 

feed items[4] = Title: Improve Teamwork with Org and Team Charts
Published: 5/20/2010 12:12:08 PM
URL: http://www.youtube.com/watch?v=PBbAzdSOn7w&feature=youtube_gdata
Author: smartdraw
Id: tag:youtube.com,2008:video:PBbAzdSOn7w
Content: 
Categories: 

feed items[5] = Title: Organizing Your Ideas with Mind Maps
Published: 5/31/2010 10:29:12 AM
URL: http://www.youtube.com/watch?v=Amzipn5xw7s&feature=youtube_gdata
Author: smartdraw
Id: tag:youtube.com,2008:video:Amzipn5xw7s
Content: 
Categories: 

feed items[6] = Title: How to Create Compelling Charts and Graphs with SmartDraw
Published: 5/30/2010 4:44:45 AM
URL: http://www.youtube.com/watch?v=xvTCQLeV7qU&feature=youtube_gdata
Author: smartdraw
Id: tag:youtube.com,2008:video:xvTCQLeV7qU
Content: 
Categories: 

feed items[7] = Title: SmartDraw Vs Visio
Published: 5/31/2010 6:49:00 AM
URL: http://www.youtube.com/watch?v=L7JVmhxRZp8&feature=youtube_gdata
Author: smartdraw
Id: tag:youtube.com,2008:video:L7JVmhxRZp8
Content: 
Categories: 

feed items[8] = Title: Introduction to SmartDraw 2010 Upgrade
Published: 5/31/2010 11:22:06 AM
URL: http://www.youtube.com/watch?v=-PWqMCWiCpQ&feature=youtube_gdata
Author: smartdraw
Id: tag:youtube.com,2008:video:-PWqMCWiCpQ
Content: 
Categories: 

feed items[9] = Title: New SmartDraw 2010 Timesavers
Published: 5/30/2010 4:04:22 AM
URL: http://www.youtube.com/watch?v=fDv-9x83GTI&feature=youtube_gdata
Author: smartdraw
Id: tag:youtube.com,2008:video:fDv-9x83GTI
Content: 
Categories: 

feed items[10] = Title: Managing Time & Resources with SmartDraw 2010
Published: 3/30/2010 5:48:42 PM
URL: http://www.youtube.com/watch?v=VSDxWqkwVIw&feature=youtube_gdata
Author: smartdraw
Id: tag:youtube.com,2008:video:VSDxWqkwVIw
Content: 
Categories: 

feed items[11] = Title: Create Killer Presentations with SmartDraw 2010
Published: 5/30/2010 8:56:00 AM
URL: http://www.youtube.com/watch?v=60qq_24JaQE&feature=youtube_gdata
Author: smartdraw
Id: tag:youtube.com,2008:video:60qq_24JaQE
Content: 
Categories: 

feed items[12] = Title: Collaborate with SmartDraw 2010
Published: 2/5/2010 12:51:15 PM
URL: http://www.youtube.com/watch?v=5-zCjjoyk1k&feature=youtube_gdata
Author: smartdraw
Id: tag:youtube.com,2008:video:5-zCjjoyk1k
Content: 
Categories: 

feed items[13] = Title: Automatically Create a Complete PowerPoint Slideshow
Published: 5/30/2010 8:17:24 AM
URL: http://www.youtube.com/watch?v=GEwEY8Ackv0&feature=youtube_gdata
Author: smartdraw
Id: tag:youtube.com,2008:video:GEwEY8Ackv0
Content: 
Categories: 

feed items[14] = Title: Annotate a Map
Published: 5/28/2010 10:33:28 AM
URL: http://www.youtube.com/watch?v=VJfNOjl3Frg&feature=youtube_gdata
Author: smartdraw
Id: tag:youtube.com,2008:video:VJfNOjl3Frg
Content: 
Categories: 

feed items[15] = Title: Easily Create 3D Charts
Published: 5/24/2010 2:22:27 PM
URL: http://www.youtube.com/watch?v=nJjrZJp7YEM&feature=youtube_gdata
Author: smartdraw
Id: tag:youtube.com,2008:video:nJjrZJp7YEM
Content: 
Categories: 

feed items[16] = Title: Control the Sequencing Order of Your Diagrams
Published: 5/16/2010 2:29:40 PM
URL: http://www.youtube.com/watch?v=zex3BhjI5Kk&feature=youtube_gdata
Author: smartdraw
Id: tag:youtube.com,2008:video:zex3BhjI5Kk
Content: 
Categories: 

feed items[17] = Title: Link Job Descriptions to Org Charts
Published: 5/29/2010 7:10:58 PM
URL: http://www.youtube.com/watch?v=TOlQjH0brrc&feature=youtube_gdata
Author: smartdraw
Id: tag:youtube.com,2008:video:TOlQjH0brrc
Content: 
Categories: 

feed items[18] = Title: Automatically Develop a Sitemap
Published: 4/16/2010 4:19:31 AM
URL: http://www.youtube.com/watch?v=vEourCP4GSg&feature=youtube_gdata
Author: smartdraw
Id: tag:youtube.com,2008:video:vEourCP4GSg
Content: 
Categories: 

feed items[19] = Title: Export Mind Maps into a Plain text Outline Format
Published: 5/23/2010 2:27:45 PM
URL: http://www.youtube.com/watch?v=YR6aOycl0pc&feature=youtube_gdata
Author: smartdraw
Id: tag:youtube.com,2008:video:YR6aOycl0pc
Content: 
Categories: 

feed items[20] = Title: Easily Email SmartDraw Diagrams to Others
Published: 4/29/2010 12:59:07 AM
URL: http://www.youtube.com/watch?v=IjgJGCOfXTc&feature=youtube_gdata
Author: smartdraw
Id: tag:youtube.com,2008:video:IjgJGCOfXTc
Content: 
Categories: 

feed items[21] = Title: Create and Print a Poster
Published: 5/25/2010 9:24:04 AM
URL: http://www.youtube.com/watch?v=0n_a4zdcBLg&feature=youtube_gdata
Author: smartdraw
Id: tag:youtube.com,2008:video:0n_a4zdcBLg
Content: 
Categories: 

feed items[22] = Title: Customize Your Gantt Charts
Published: 5/20/2010 2:37:18 PM
URL: http://www.youtube.com/watch?v=x2Wg5p4FTXA&feature=youtube_gdata
Author: smartdraw
Id: tag:youtube.com,2008:video:x2Wg5p4FTXA
Content: 
Categories: 

feed items[23] = Title: Capture and Annotate a Webpage
Published: 5/17/2010 2:10:36 PM
URL: http://www.youtube.com/watch?v=LkO7WOMqg2M&feature=youtube_gdata
Author: smartdraw
Id: tag:youtube.com,2008:video:LkO7WOMqg2M
Content: 
Categories: 

feed items[24] = Title: Export to PDF in One Click with SmartDraw
Published: 4/1/2010 9:03:25 AM
URL: http://www.youtube.com/watch?v=hmmaIU3hJbo&feature=youtube_gdata
Author: smartdraw
Id: tag:youtube.com,2008:video:hmmaIU3hJbo
Content: 
Categories: 

